assets/js/components/ReportErrorActions.js

```diff
@@ -0,0 +1,132 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Button } from 'googlesitekit-components';
+import { CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+import {
+	isErrorRetryable,
+	isInsufficientPermissionsError,
+} from '../util/errors';
+import useViewOnly from '../hooks/useViewOnly';
+import Link from './Link';
+
+const { useSelect, useDispatch } = Data;
+
+export default function ReportErrorActions( { moduleSlug, error } ) {
+	const isViewOnly = useViewOnly();
+	const storeName = useSelect( ( select ) =>
+		select( CORE_MODULES ).getModuleStoreName( moduleSlug )
+	);
+	const requestAccessURL = useSelect( ( select ) =>
+		typeof select( storeName )?.getServiceEntityAccessURL === 'function'
+			? select( storeName ).getServiceEntityAccessURL()
+			: null
+	);
+
+	const errors = Array.isArray( error ) ? error : [ error ];
+
+	const retryableErrors = errors.filter(
+		( err ) =>
+			isErrorRetryable( err, err.selectorData ) &&
+			err.selectorData.name === 'getReport'
+	);
+
+	const showRetry = !! retryableErrors.length && ! isViewOnly;
+
+	const errorTroubleshootingLinkURL = useSelect( ( select ) => {
+		const err = {
+			...( showRetry ? retryableErrors[ 0 ] : errors[ 0 ] ),
+		};
+
+		if ( isInsufficientPermissionsError( err ) ) {
+			err.code = `${ moduleSlug }_insufficient_permissions`;
+		}
+
+		return select( CORE_SITE ).getErrorTroubleshootingLinkURL( err );
+	} );
+
+	const dispatch = useDispatch();
+
+	const hasInsufficientPermissionsError = errors.some( ( err ) =>
+		isInsufficientPermissionsError( err )
+	);
+
+	const handleRetry = useCallback( () => {
+		retryableErrors.forEach( ( err ) => {
+			const { selectorData } = err;
+			dispatch( selectorData.storeName ).invalidateResolution(
+				selectorData.name,
+				selectorData.args
+			);
+		} );
+	}, [ dispatch, retryableErrors ] );
+
+	const showRequestAccessURL =
+		requestAccessURL && hasInsufficientPermissionsError && ! isViewOnly;
+
+	return (
+		<div className="googlesitekit-report-error-actions">
+			{ showRequestAccessURL && (
+				<Button href={ requestAccessURL } target="_blank">
+					{ __( 'Request access', 'google-site-kit' ) }
+				</Button>
+			) }
+			{ showRetry ? (
+				<Fragment>
+					<Button onClick={ handleRetry }>
+						{ __( 'Retry', 'google-site-kit' ) }
+					</Button>
+					<span className="googlesitekit-error-retry-text">
+						{ __( 'Retry didn’t work?', 'google-site-kit' ) }{ ' ' }
+					</span>
+					<Link href={ errorTroubleshootingLinkURL } external>
+						{ __( 'Get help', 'google-site-kit' ) }
+					</Link>
```

Comment:
While we're here, let's improve this with `createInterpolateElement`, eg.

```suggestion
					<span className="googlesitekit-error-retry-text">
						{ createInterpolateElement( 
							__( 'Retry didn’t work? <HelpLink />', 'google-site-kit' ),
							{
								HelpLink: (
									<Link href={ errorTroubleshootingLinkURL } external>
										{ __( 'Get help', 'google-site-kit' ) }
									</Link>
								)
							}
						) }
					</span>
```